src/module_wrap.cc

```diff
@@ -543,11 +543,51 @@ const PackageConfig& GetPackageConfig(Environment* env,
     main_std.assign(std::string(*main_utf8, main_utf8.length()));
   }
 
+  Local<Value> pkg_mode_v;
+  PackageMode pkg_mode = CJS;
+  std::string pkg_mode_std;
+  if (pkg_json->Get(env->context(), env->mode_string()).ToLocal(&pkg_mode_v)) {
+    Utf8Value pkg_mode_utf8(isolate, pkg_mode_v);
+    pkg_mode_std.assign(std::string(*pkg_mode_utf8, pkg_mode_utf8.length()));
+    if (pkg_mode_std == "esm") {
+      pkg_mode = ESM;
+    }
+  }
+
   auto entry = env->package_json_cache.emplace(path,
-      PackageConfig { Exists::Yes, IsValid::Yes, has_main, main_std });
+      PackageConfig { Exists::Yes, IsValid::Yes, has_main,
+                      main_std, pkg_mode });
   return entry.first->second;
 }
 
+PackageMode GetPackageMode(Environment* env, const URL& search) {
+  URL pjsonPath("package.json", &search);
```

Comment:
Style: `pjson_path` (i.e. snake_case, not camelCase.). Ditto on line 571, 581, 754, etc.